test/units/testsuite-56.sh

```diff
@@ -24,13 +24,13 @@ chmod +x /tmp/test56-exit-cgroup.sh
 
 # service should be stopped cleanly
 systemd-run --wait --unit=one -p Type=notify -p ExitType=cgroup \
-    -p ExecStartPost='bash -c "systemctl stop one &"' \
+    -p ExecStartPost='bash -c "(sleep 1; systemctl stop one) &"' \
```

Comment:
What is this trying to do? Ensuring that the stop is sent only after the unit is running? But for that it doesn't need to be in ExecStartPost, the --wait can be removed and it can be done as a separate command. systemd-run will not return until --ready has been sent. Or am I missing something?

Reply:
no, it's ensuring that the stop is sent only after the ExecStartPost command itself (bash) is finished. Otherwise the service never reaches `running` state and ends up in ` stop-sigterm` instead, which causes the whole unit to fail because the control process never gets a chance to exit properly as far as systemd knows

Reply:
Why is the ExecStartPost relevant here? If it wasn't, it would reach the running state after the sd_notify call, no?

Reply:
no, after the sd_notify call the service goes to the `start-post` status, and only reaches `running` after the `ExecStartPost` command finishes

Reply:
And if there's no ExecStartPost?

Reply:
You mean run ` systemctl stop one` directly from the script itself? that should also work, but then for `four.service` and `five.service` it doesn't demonstrate that the main process actually already exited

Reply:
It would most of the time, since the sd-notify is the last thing and then it exits (you can also add `--no-block`). There is a chance of a false positive, but that's better than a false negative, which is very time consuming